Fix theme palette management

willgearty opened the pull request:
This fixes the management of a custom theme palette on the `/themes/customize` page. Since we can't change what the theme colors are, I've separated them out and now we only save the "Custom Palette" that is defined by the user:
![image](https://user-images.githubusercontent.com/7232514/128203411-2917b935-d592-45be-a851-c4f0402d6564.png)

If you somehow specify a color that is already in the theme palette or a duplicate of a color already in your custom palette, it's automatically removed. Also, you can now remove any colors from your palette (not just the last one) by clicking on them then clicking the "Remove" button:
![image](https://user-images.githubusercontent.com/7232514/128203733-387d7999-2903-4e01-ac33-10d26ac729ae.png)

I also updated the colorpicker (v. 1 -> v. 2), which is a little more visually appealing. While I was at it, I updated the font size and family pickers so you can actually see what the different font sizes and families look like when you are picking them:
![image](https://user-images.githubusercontent.com/7232514/128203959-ed872c31-c23d-4892-87a7-0603feb4af5e.png)

Fixes https://github.com/learning-unlimited/ESP-Website/issues/2907.

kkbrum reviewed (changes requested):
This all looks good, but while you're here...

- [x] Can you fix the divs so the customize theme page isnt cut off at the bottom in fruitsalad
- [ ] If you try to save a custom theme with a name that includes slashes, it breaks. Can we fail with an informative error at least or escape things so it doesnt fail?
- [ ] This is probably a known larger thing, but if I change the font, it changes in only very few places?

willgearty commented:
I've fixed the first two items, but the latter is definitely outside of the scope of this PR. It's also already been documented in https://github.com/learning-unlimited/ESP-Website/issues/2906 and https://github.com/learning-unlimited/ESP-Website/issues/1342. I've added them both to SR14.

kkbrum reviewed (approved):
Extraordinary!
